src/transformers/pipelines/conversational.py

```diff
@@ -246,18 +259,7 @@ def __call__(self, conversations: Union[Conversation, List[Conversation]], num_w
         return outputs
 
     def preprocess(self, conversation: Conversation, min_length_for_response=32) -> Dict[str, Any]:
-        if not isinstance(conversation, Conversation):
-            raise ValueError("ConversationalPipeline, expects Conversation as inputs")
-        if conversation.new_user_input is None:
-            raise ValueError(
-                f"Conversation with UUID {type(conversation.uuid)} does not contain new user input to process. "
-                "Add user inputs with the conversation's `add_user_input` method"
-            )
-        if hasattr(self.tokenizer, "_build_conversation_input_ids"):
-            input_ids = self.tokenizer._build_conversation_input_ids(conversation)
-        else:
-            # If the tokenizer cannot handle conversations, we default to only the old version
-            input_ids = self._legacy_parse_and_tokenize(conversation)
+        input_ids = self.tokenizer.apply_chat_template(conversation)
```

Comment:
for the pipeline I think it makes sense to cache the template format as you have more fredom to recompute it. In general with a setter and a `_template` you could have this pretty easily no?

Reply:
I added a cacheing method to the base tokenizer for this! I can also factor it out of the tokenizer as a function if you think that's cleaner. The tokenizer also retains the Jinja env as `self._jinja_env` after the first time so it doesn't have to reinitialize it every time.